### PR TITLE
Port to CMake

### PR DESCRIPTION
**Pull request by PureTryOut**
This requires some changes to mapbox-gl-native as well to be a proper CMake using citizen, https://github.com/mapbox/mapbox-gl-native/pull/16591

### CODE REVIEW

**Comment by rinigus** [comment]
`set(QT_MIN_VERSION "5.9.0")` will not work - SFOS is on 5.6

Can't test it without QMapboxGL changes. :(

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		PkgConfig::OPENSSL)
+endif()
+
+FindQtInstallQml()
+
+install(FILES 
+	plugin/MapboxMapGestureArea.qml
+	plugin/qmldir
+	DESTINATION ${QT_INSTALL_QML}/MapboxMap/)
+install(TARGETS qmlmapboxplugin DESTINATION ${QT_INSTALL_QML}/MapboxMap/)
```
Comment (rinigus) [inline src/CMakeLists.txt:53]: I guess it should be `qmlmapboxglplugin`

Comment (PureTryOut) [reply src/CMakeLists.txt:53]: I already changed that on my desktop but seems I forgot to push :unamused:

**Comment by rinigus** [comment]
Thank you very much, merging!

**Pull request merged**